pkgs/development/tools/boost-build/default.nix

```diff
@@ -36,10 +36,10 @@ stdenv.mkDerivation {
   patches = useBoost.boostBuildPatches or []
     ++ lib.optional (useBoost ? version && lib.versionAtLeast useBoost.version "1.81") ./fix-clang-target.patch;
 
-  # Upstream defaults to gcc on darwin, but we use clang.
   postPatch = ''
-    substituteInPlace src/build-system.jam \
-    --replace "default-toolset = darwin" "default-toolset = clang-darwin"
+    # Upstream uses arm64, but nixpkgs uses aarch64.
+    substituteInPlace src/tools/clang.jam \
+      --replace-fail 'arch = arm64' 'arch = aarch64'
```

Comment:
I wonder if we should consider adding aliases for this to our LLVM build at some point…

Reply:
seems like the replace-fail should be replace-warn as older boosts are breaking here. and think that the default-toolset should be added back due to older boosts as well, using `replace-warn`.

Reply:
see https://github.com/NixOS/nixpkgs/pull/350658